Regular Expressions/RegExp (Basic).sublime-syntax

```diff
@@ -90,84 +86,123 @@ contexts:
       pop: 1
 
   groups:
-    - include: group-comments
     - match: \(
       scope: keyword.control.group.regexp
-      push: group-start
+      push: [group-body-placeholder, maybe-unexpected-quantifiers, group-start]
 
   groups-extended:
-    - include: group-comments
     - match: \(
       scope: keyword.control.group.regexp
-      push: group-start-extended
+      push: [group-body-extended-placeholder, maybe-unexpected-quantifiers, group-start-extended]
+
+  group-body-placeholder:
+    - match: ''
+      set: group-body
+
+  group-body-extended-placeholder:
+    - match: ''
+      set: group-body-extended
 
   group-start:
     - meta_scope: meta.group.regexp
-    - match: \?(<[=!]|[>=:!])
-      scope: constant.other.assertion.regexp
-      set: [group-body, maybe-unexpected-quantifiers]
-    # Activates 'x' mode
-    - match: (\?[ims]*x[ixms]*(?:-[ims]+)?)(\))
-      captures:
-        1: storage.modifier.mode.regexp
-        2: keyword.control.group.regexp
-      set: [base-group-extended, maybe-unexpected-quantifiers]
-    # Groups with 'x' mode
-    - match: '\?[ims]*x[ixms]*(?:-[ims]+)?:'
-      scope: storage.modifier.mode.regexp
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-    # Other modifiers
-    - match: '\?(?:[ixms]*-)?[ixms]+:'
-      scope: storage.modifier.mode.regexp
-      set: [group-body, maybe-unexpected-quantifiers]
     - include: group-start-common
-    - match: ''
-      set: [group-body, maybe-unexpected-quantifiers]
 
   group-start-extended:
     - meta_scope: meta.group.extended.regexp
+    - include: group-start-common
+
+  group-start-common:
```

Comment:
Can't make a github suggestion for such a large block, but how about...

```
  groups:
    - match: \(
      scope: meta.group.regexp keyword.control.group.regexp
      push: [group-body, maybe-unexpected-quantifiers, group-start]

  groups-extended:
    - match: \(
      scope: meta.group.extended.regexp keyword.control.group.regexp
      push: [group-body-extended, maybe-unexpected-quantifiers, group-start]

  group-start:
     - match: \?(<[=!]|[>=:!])
```

By adding the meta scope to the opening `\(` match it appears to be possible to remove some of those intermediate contexts. Note `group-start-common` becomes `group-start`.